### DISCUSSION TITLE
Change url display in nextjs as [slug].html

### DISCUSSION THREAD

**Discussion by thainv92** [vercel/next.js#59639] [Help]
### Summary

I have a route /product/[slug] that currently displays the URL in the browser as: https://navata.com/product/television. However, for better SEO, I'd prefer it to display as https://navata.com/television.html. 
I'm using Next.js, but I'm unsure how to implement this change. 
Could you please help me resolve this issue?

**Comment by icyJoseph** [comment]
I think there's two parts to this, first, you want to configure your server to server using `.html` extensions, so that when people arrive to your site `/television`, the server redirects them to `/television.html`. Might be worth skipping this though, today not that many use this kind of approach, but it is doable.

You also need to implement rewrites, at next.config.js level, or middleware, to serve `/product/[slug]` when `/[slug]` is received, you'd need to also validate slugs and such:

- https://nextjs.org/docs/app/api-reference/next-config-js/rewrites
